snuba/datasets/entities/discover.py

```diff
@@ -317,39 +328,83 @@ def __init__(self) -> None:
-                            )
+        events_pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SelectedStorageQueryPlanBuilder(
+                selector=EventsQueryStorageSelector(
+                    mappers=events_translation_mappers.concat(
+                        transaction_translation_mappers
+                    )
+                    .concat(null_function_translation_mappers)
+                    .concat(
+                        TranslationMappers(
+                            # XXX: Remove once we are using errors
+                            columns=[
+                                ColumnToMapping(
+                                    None, "release", None, "tags", "sentry:release"
+                                ),
+                                ColumnToMapping(
+                                    None, "dist", None, "tags", "sentry:dist"
+                                ),
+                                ColumnToMapping(
+                                    None, "user", None, "tags", "sentry:user"
+                                ),
+                            ],
+                            subscriptables=[
+                                SubscriptableMapper(None, "tags", None, "tags"),
+                                SubscriptableMapper(None, "contexts", None, "contexts"),
+                            ],
                         )
                     )
+                )
+            ),
+        )
+
+        discover_storage = get_storage(StorageKey.DISCOVER)
+
+        discover_pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=discover_storage,
+                mappers=events_translation_mappers.concat(
+                    transaction_translation_mappers
+                )
+                .concat(null_function_translation_mappers)
+                .concat(
+                    TranslationMappers(
+                        subscriptables=[
+                            SubscriptableMapper(None, "tags", None, "tags"),
+                            SubscriptableMapper(None, "contexts", None, "contexts"),
+                        ],
+                    )
                 ),
+            )
+        )
+
+        def selector_func(_query: Query) -> Tuple[str, List[str]]:
+            if random.random() < float(
+                state.get_config("discover_query_percentage", 0)
+            ):
+                return "events", ["discover"]
+
+            return "events", []
+
+        def callback_func(results: List[Tuple[str, QueryResult]]) -> None:
+            _, primary_result = results.pop(0)
+
+            for _, result in results:
+                if result.result["data"] == primary_result.result["data"]:
+                    metrics.increment("query_result", tags={"match": "true"})
+                else:
+                    metrics.increment("query_result", tags={"match": "false"})
```

Comment:
Is this logging something to debug with as well? Like if we find that there are mismatches how do we determine what the actual queries are that aren't matching?

Reply:
We'll do a more sophisticated comparison of results later down the line (ideally when consumers are synced), for now this is mostly just to ensure we can actually run queries on the Discover table without errors.